src/control/lib/control/pool.go

```diff
@@ -344,7 +343,8 @@ type (
 	// PoolQueryReq contains the parameters for a pool query request.
 	PoolQueryReq struct {
 		poolRequest
-		ID string
+		ID      string
+		InfoBit int32
```

Comment:
The control API should be independent of the constants from the C code. We can (and IMO should) simplify the interface presented to the caller.

```suggestion
		IncludeDisabledRanks bool
```

I don't know that it's necessary for the control plane to know about the bits at all. If we set up the protobuf version of the request to have booleans like this, or its own independent bitmask, we can translate it over to these DPI_* codes once we get into the C code.

Reply:
Worth noting it also seems like the engines_enabled bit is the only one we care about, the rest are automatically chosen.

Reply:
The idea was to stick to the C code to easily be able to manage the evolution of this bitfield and to not add to much extra field to the RPC message.